File: app/Reco3D/bin/run_reco.cxx
#include <iostream>
#include <vector>
#include <string>
#include <fstream>

#include "Processor/ProcessDriver.h"
#include "ReadJarrettFile.h"

int main(int nargs, char** argv){

    if(nargs < 3){
        std::cout << "usage : ./run_reco [cfg file] [input 1] [input 2] ..." << std::endl;
        return 1;
    }

    std::cout << "***********************" << std::endl;
    std::cout << "  RUN 3D RECO : START  " << std::endl;
    std::cout << "***********************" << std::endl;

    std::string cfg = argv[1];
    std::vector<std::string> data_inputs;
    for(int i = 2; i < nargs; i++){data_inputs.push_back(argv[i]);}

    larcv::ProcessDriver proc("ProcessDriver");
    proc.configure(cfg);
    proc.override_input_file(data_inputs);

    //larcv::ProcessID_t algo = proc.process_id("ReadJarrettFile");
    //const larcv::ProcessBase *algo_id = proc.process_ptr(algo);
    //algo_id->SetSplineLocation("Proton_Muon_Range_dEdx_LAr_TSplines.root");
    
    larcv::ProcessID_t algo_id = proc.process_id("ReadJarrettFile");
    larcv::ReadJarrettFile *algo = (larcv::ReadJarrettFile*)proc.process_ptr(algo_id);
    algo->SetSplineLocation("/Users/hourlier/Documents/PostDocMIT/Research/MicroBooNE/dllee_unified/LArCV/app/Reco3D/Proton_Muon_Range_dEdx_LAr_TSplines.root");

    proc.initialize();
    proc.batch_process(0);
    std::cout << "RUN 3D RECO : about to finalize" << std::endl;
    proc.finalize();

    std::cout << "**********************" << std::endl;
    std::cout << "  RUN 3D RECO : STOP  " << std::endl;
    std::cout << "**********************" << std::endl;

    return 0;

}

File: app/LArOpenCVHandle/cfg/mac/run_reco.py
import os, sys

if len(sys.argv) != 6:
    print 
    print "CFG     = str(sys.argv[1])"
    print "ANAFILE = str(sys.argv[2])"
    print "PGRFILE = str(sys.argv[3])"
    print "INFILE  = str(sys.argv[4])"
    print "OUTDIR  = str(sys.argv[5])" 
    print 
    sys.exit(1)

BASE_PATH = os.path.realpath(__file__)
BASE_PATH = os.path.dirname(BASE_PATH)
sys.path.insert(0,BASE_PATH)

CFG     = str(sys.argv[1])
ANAFILE = str(sys.argv[2])
PGRFILE = str(sys.argv[3])
INFILE  = str(sys.argv[4])
OUTDIR  = str(sys.argv[5])

from larcv import larcv
import ROOT
from ROOT import std

proc = larcv.ProcessDriver('ProcessDriver')
proc.configure(CFG)
flist=ROOT.std.vector('std::string')()
flist.push_back(ROOT.std.string(INFILE))
proc.override_input_file(flist)
proc.override_ana_file(os.path.join(OUTDIR,ANAFILE))
proc.override_output_file(os.path.join(OUTDIR,PGRFILE))
proc.initialize()
proc.batch_process()
proc.finalize()

sys.exit(0)
